File: rps.java
import java.util.Random;
import java.util.Scanner;

public class rps {
    public static void main(String[] args) {
        System.out.println("==========================================");
        System.out.println("Rock, paper and scissors...");
        System.out.println("==========================================");
        // 0 = rock
        // 1 = paper
        // 2 = scissors
        Scanner sc = new Scanner(System.in);
        Random rand = new Random();
        int compCount = 0;
        int playerCount = 0;
        System.out.println("How many rounds would you like to play?");
        System.out.print(">> ");
        int n = sc.nextInt();
        for (int i = 1; i <= n; i++) {
            int num = rand.nextInt(3);
            // User Input
            System.out.print("Press 'r' for rock, 'p' for paper and 's' for scissors: ");
            char pick = sc.next().charAt(0);
            // Displaying Computer's input
            if (pick == 'r' || pick == 'p' || pick == 's') {
                if (num == 0) {
                    System.out.println("Computer's pick = rock");
                } else if (num == 1) {
                    System.out.println("Computer's pick = paper");
                } else {
                    System.out.println("Computer's pick = scissor");
                }
                // Displaying Player's input
                if (pick == 'r') {
                    System.out.println("Your pick = rock");
                } else if (pick == 'p') {
                    System.out.println("Your pick = paper");
                } else if (pick == 's') {
                    System.out.println("Your pick = scissor");
                }
            }
            // If user enters something else (not r,p or s)
            else {
                System.out.println("----------------------------");
                System.out.println("Invalid option entered. Try again!");
                i--;
                System.out.println("----------------------------");
            }
            // When Computer picks rock
            if (num == 0 && pick == 'r') {
                System.out.println("----------------------------");
                System.out.println("Round Tied!");
                System.out.println("----------------------------");
            } else if (num == 0 && pick == 'p') {
                System.out.println("----------------------------");
                System.out.println("Player wins the round!");
                System.out.println("----------------------------");
                playerCount++;
            } else if (num == 0 && pick == 's') {
                System.out.println("----------------------------");
                System.out.println("Computer wins the round!");
                System.out.println("----------------------------");
                compCount++;
            }
            // When computer picks paper
            if (num == 1 && pick == 'p') {
                System.out.println("----------------------------");
                System.out.println("Round Tied!");
                System.out.println("----------------------------");
            } else if (num == 1 && pick == 's') {
                System.out.println("----------------------------");
                System.out.println("Player wins the round!");
                System.out.println("----------------------------");
                playerCount++;
            } else if (num == 1 && pick == 'r') {
                System.out.println("----------------------------");
                System.out.println("Computer wins the round!");
                System.out.println("----------------------------");
                compCount++;
            }
            // When computer picks scissors
            if (num == 2 && pick == 's') {
                System.out.println("----------------------------");
                System.out.println("Round Tied!");
                System.out.println("----------------------------");
            } else if (num == 2 && pick == 'r') {
                System.out.println("----------------------------");
                System.out.println("Player wins the round!");
                System.out.println("----------------------------");
                playerCount++;
            } else if (num == 2 && pick == 'p') {
                System.out.println("----------------------------");
                System.out.println("Computer wins the round!");
                System.out.println("----------------------------");
                compCount++;
            }
        }
        // Results
        System.out.println("==========================");
        System.out.println("End of " + n + " rounds...");
        System.out.println("==========================");
        System.out.println("Final Score:");
        System.out.println("Computer's points = " + compCount);
        System.out.println("Player's points = " + playerCount);
        System.out.println("**************************");
        if (compCount == playerCount) {
            System.out.println("Game tied!");
        } else if (compCount > playerCount) {
            System.out.println("Computer won the game!");
        } else {
            System.out.println("Player won the game!");
        }
        System.out.println("**************************");
        sc.close();
    }
}

File: rps.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
void main()
{
    int playerScore = 0, computerScore = 0;
    srand(time(0));
    int random = rand() % 3 + 1;
    // rock=1
    // paper=2
    // scissors=3
    int rounds;
    printf("How many rounds would you like to play?\n>> ");
    scanf("%d", &rounds);
    for (int i = 1; i <= rounds; i++)
    {
        printf("Press 'r' for rock, 'p' for paper, and 's' for scissors\n>> ");
        char playerPick;
        scanf("%c", &playerPick);
        scanf("%c", &playerPick);
        if (playerPick == 'r' || playerPick == 'p' || playerPick == 's')
        {
            // When computer's pick is rock
            if (random == 1 && playerPick == 'r')
            {
                printf("Player's pick = rock\n");
                printf("Computer's pick = rock\n");
                printf("----------------------------\n");
                printf("Round Tied!\n");
                printf("----------------------------\n");
            }
            else if (random == 1 && playerPick == 'p')
            {
                printf("Player's pick = paper\n");
                printf("Computer's pick = rock\n");
                printf("----------------------------\n");
                printf("Player wins the round!\n");
                printf("----------------------------\n");
                playerScore++;
            }
            else if (random == 1 && playerPick == 's')
            {
                printf("Player's pick = scissors\n");
                printf("Computer's pick = rock\n");
                printf("----------------------------\n");
                printf("Computer wins the round!\n");
                printf("----------------------------\n");
                computerScore++;
            }

            // When computer's pick is paper
            if (random == 2 && playerPick == 'r')
            {
                printf("Player's pick = rock\n");
                printf("Computer's pick = paper\n");
                printf("----------------------------\n");
                printf("Computer wins the round!\n");
                printf("----------------------------\n");
                computerScore++;
            }
            else if (random == 2 && playerPick == 'p')
            {
                printf("Player's pick = paper\n");
                printf("Computer's pick = paper\n");
                printf("----------------------------\n");
                printf("Round Tied!\n");
                printf("----------------------------\n");
            }
            else if (random == 2 && playerPick == 's')
            {
                printf("Player's pick = scissors\n");
                printf("Computer's pick = paper\n");
                printf("----------------------------\n");
                printf("Player wins the round!\n");
                printf("----------------------------\n");
                playerScore++;
            }

            // When computer's pick is scissors
            if (random == 3 && playerPick == 'r')
            {
                printf("Player's pick = rock\n");
                printf("Computer's pick = scissors\n");
                printf("----------------------------\n");
                printf("Player wins the round!\n");
                printf("----------------------------\n");
                playerScore++;
            }
            else if (random == 3 && playerPick == 'p')
            {
                printf("Player's pick = paper\n");
                printf("Computer's pick = scissors\n");
                printf("----------------------------\n");
                printf("Computer wins the round!\n");
                printf("----------------------------\n");
                computerScore++;
            }
            else if (random == 3 && playerPick == 's')
            {
                printf("Player's pick = scissors\n");
                printf("Computer's pick = scissors\n");
                printf("----------------------------\n");
                printf("Round Tied!\n");
                printf("----------------------------\n");
            }
        }
        else
        {
            printf("----------------------------\n");
            printf("Invalid option entered. Try again.\n");
            printf("----------------------------\n");
            i--;
        }
    }
    printf("==========================\n");
    printf("End of %d rounds\n", rounds);
    printf("==========================\n");
    printf("Final Scores:\n");
    printf("Player's score = %d\n", playerScore);
    printf("Computer's score = %d\n", computerScore);
    printf("**************************\n");
    if (playerScore > computerScore)
    {
        printf("Player wins the round!\n");
    }
    else if (playerScore < computerScore)
    {
        printf("Computer wins the round!\n");
    }
    else
    {
        printf("Match Tied!\n");
    }
}
